src/custom-element.js

```diff
@@ -105,6 +108,16 @@ function createBaseCustomElementClass(win) {
    * @abstract @extends {HTMLElement}
    */
   class BaseCustomElement extends htmlElement {
+    /**
+     * Part of CE API. Enumerates the set of attributes that are notified to
+     * the element.
+     * @return {!Array<string>}
+     * @final
+     */
+    static get observedAttributes() {
+      return MEDIA_ATTRS;
```

Comment:
My CE polyfill does not do `observedAttributes`. We'll need to add that if we want to use this.

Reply:
Actually, I think it won't be end of the world if it didn't work on older platforms. And, also, we can separate media/sizes/heights attribute changes from this pull request. The legacy code didn't support these changes. WDYT?

Reply:
Per this thread, I removed mutation support. This code is now provides the same guarantees as the old code, just more targeted updates based on the events.